sktime/forecasting/compose/_reduce.py

```diff
@@ -264,6 +265,76 @@ def _predict_in_sample(self, fh, X=None, return_pred_int=False, alpha=None):
         raise NotImplementedError("in-sample predictions are not implemented")
 
 
+class _MultioutputReducer(_RequiredForecastingHorizonMixin, BaseReducer):
+    strategy = "multioutput"
+
+    def fit(self, y, X=None, fh=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : pd.Series
+            Target time series to which to fit the forecaster.
+        fh : int, list or np.array, optional (default=None)
+            The forecasters horizon with the steps ahead to to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables are ignored
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        self._set_y_X(y, X)
+        if X is not None:
+            raise NotImplementedError()
+        self._set_fh(fh)
+        if len(self.fh.to_in_sample(self.cutoff)) > 0:
+            raise NotImplementedError("In-sample predictions are not implemented")
+
+        self.step_length_ = check_step_length(self.step_length)
+        self.window_length_ = check_window_length(self.window_length)
+
+        # for the multioutput reduction strategy, a single forecaster is fitted
+        # simultaneously to all the future steps in the forecasting horizon
+        # by reducing to a forecaster that can handle multi-dimensional outputs
+        self._cv = SlidingWindowSplitter(
+            fh=self.fh.to_relative(self.cutoff),
+            window_length=self.window_length_,
+            step_length=self.step_length_,
+            start_with_window=True,
+        )
+
+        # transform data using rolling window split
+        X, Y_train = self._transform(y, X)
+
+        # fit regressor to training data
+        self.regressor.fit(X, Y_train)
```

Comment:
What's the error here if the the regressor does not support multi-output? Is it user friendly or should we include some kind of check on the regressor if it is compatible with the multi output strategy?

Reply:
I do not know, nor do I know how I would find out. I am trusting the error is user-friendly...

1. How do I find a regressor that does not support multi-output? I searched on google and somebody asked and got no reponse. https://datascience.stackexchange.com/questions/25829/scikit-learn-which-regressors-natively-support-multi-target-regression
2. How would we check if a regressor is compatible?

Reply:
I think ultimately we can only have heuristics for finding out whether an estimator supports multiouput or not. One thing to look for is the multi output mixin class or multiouput tag from scikit-learn: https://github.com/scikit-learn/scikit-learn/blob/95119c13af77c76e150b753485c662b7c52a41a2/sklearn/base.py#L758

Reply:
to add a detail: if a `sklearn` estimator does not support multioutput, you can always make it support it by wrapping it in `MultiOutputRegressor` or `MultiOutputClassifier` (a kind of reduction/composition).

Reply:
Responding to the detail, if they estimator does not support multioutput, they should just use direct!

Reply:
@Lovkush-A yes, wrapping in `MultiOutputRegressor` is the same as using `direct` - perhaps something we should add to the docstring. Similarly, there is [`RegressorChain`](https://scikit-learn.org/stable/modules/generated/sklearn.multioutput.RegressorChain.html) for recursive (or direct?) strategy.

Reply:
I am not sure if docstring is correct place for that. Feels like the kind of thing that should go in the user-guide.

Reply:
hm, this starts souding like an inconsistency!

I.e., if the wrapped estimator supports multi-output that is not identical with looping over output variables, what should the `direct` strategy do?

I think there are actually two direct strategies!
* `direct` - loops over the forecasting horizon, using single-output versions of the wrapped estimator; fits multiple single-output versions of the estimator internally
* `direct-multioutput` - uses the wrapped estimator's multi-output mode; fits a single multi-output estimator internally

While `direct-multioutput` after applying `MultiOutputRegressor` is always the same as `direct`, it is not true that `direct-multioutput` is always the same as `direct`, hence these are two different strategies for reduction!

Reply:
@mloning implementing it this way, i.e., properly splitting up the two reduction strategies here, would also remove the need to check whether the estimator supports multi-output. The docstring can say that multi-output is assumed to be supported if you use `direct-multioutput` as your reduction strategy.

Reply:
I do not follow. Your `direct-multioutput` sounds like `multioutput`.

Reply:
ok, I'm confused now, probably entirely my fault ... what are the current options for strategies?
And which option did your original comment refer to ("if not support multioutput")?

Reply:
We have three strategies. `direct`, `recursive` and (in this PR) `multioutput`. 

Markus was interested in knowing what error message arises if person X tries to use the multioutput strategy with a regressor R that does not support multioutput. You then suggested that X could wrap the regressor R using sklearn's `MultiOutputRegressor`, so you'd get something like: `MultioutputReductionForecaster(MultiOutputRegressor(R),...)`

I then said that you do not need this multitoutput as we can just use direct:
`DirectionReductionForecaster(R,...)`

Reply:
Get it now.
So, I'm saying: in that case (regressor does not have multi-output support), I'd *still* wrap it in `MultiOutputRegressor`.
Why: because then overall behaviour is consistent and does not raise an error when using the `multioutput` strategy

Even if in that special case `multioutput` would give you the same behaviour as `direct`, it's fine - since it's not the case in general (just in that case). Both ways get you the expected result rather than an error message which wouldn't be clear why it is raised, if the user just starts with specifying either `multioutput` or `direct`.

Reply:
Okay, I like that solution too. But the problem still remains that we need to check if the passed regressor support multioutput natively or not. And I would add a warning if it doesn't, since we change the input and wrap it automatically in the `MultioutputRegressor` we should let the user know about it.

Reply:
My opinion is that using a non-multitoutput regressor in the multioutput regression should raise an error (as it currently will). Maybe there is a principle of software design I do not know about, but I think a user should be incentivised to use the tools properly. If they want direct strategy, they should choose it, if they want multioutput, they should choose an appropriate regressor.

Reply:
I agree @Lovkush-A, that's why I think raising the warning may be a compromise. I suggest to leave it as it is for now, merge the PR and wait for some user feedback, that also saves us from having to check the estimator type.